ui/component/headerNotificationButton/view.jsx

```diff
@@ -5,37 +5,229 @@ import { ENABLE_UI_NOTIFICATIONS } from 'config';
+
+  const [anchorEl, setAnchorEl] = React.useState(null);
+  const [clicked, setClicked] = React.useState(false);
+  const open = Boolean(anchorEl);
+  const handleClick = (event) => setAnchorEl(!anchorEl ? event.currentTarget : null);
+  const handleClose = () => setAnchorEl(null);
+
+  const menuProps = {
+    id: 'notification-menu',
+    anchorEl,
+    open,
+    onClose: handleClose,
+    MenuListProps: {
+      'aria-labelledby': 'basic-button',
+      sx: { padding: 'var(--spacing-xs)' },
+    },
+    className: 'menu__list--header menu__list--notifications',
+    sx: { 'z-index': 2 },
+    PaperProps: { className: 'MuiMenu-list--paper' },
+    disableScrollLock: true,
+  };
+
+  const handleClickAway = () => {
+    if (!clicked) {
+      if (open) setClicked(true);
+    } else {
+      setAnchorEl(null);
+      setClicked(false);
+    }
+  };
+
+  const creatorIcon = (channelUrl, channelThumbnail) => (
+    <UriIndicator uri={channelUrl} link showAtSign channelInfo={{ uri: channelUrl, name: '' }}>
+      <ChannelThumbnail small thumbnailPreview={channelThumbnail} uri={channelThumbnail ? undefined : channelUrl} />
+    </UriIndicator>
+  );
 
   function handleMenuClick() {
     if (unseenCount > 0) doSeeAllNotifications();
     push(`/$/${PAGES.NOTIFICATIONS}`);
   }
 
+  function handleNotificationDelete(e, id) {
+    e.stopPropagation();
+    deleteNotification(id);
+  }
+
+  React.useEffect(() => {
+    if (!open) setClicked(false);
+  }, [open]);
+
   if (!notificationsEnabled) return null;
 
+  function handleNotificationClick(notification) {
+    const { id, notification_parameters, is_read } = notification;
+
+    if (!is_read) {
+      seeNotification([id]);
+      readNotification([id]);
+    }
+    let notificationLink = formatLbryUrlForWeb(notification_parameters.device.target);
+    if (notification_parameters.dynamic.hash) {
+      notificationLink += '?lc=' + notification_parameters.dynamic.hash + '&view=discussion';
+    }
+    push(notificationLink);
+  }
+
+  function menuEntry(notification) {
+    const { id, active_at, notification_rule, notification_parameters, is_read, type } = notification;
+
+    let channelUrl;
+    let icon;
+    switch (notification_rule) {
+      case RULE.CREATOR_SUBSCRIBER:
+        icon = <Icon icon={ICONS.SUBSCRIBE} sectionIcon />;
+        break;
+      case RULE.COMMENT:
+      case RULE.CREATOR_COMMENT:
+        channelUrl = notification_parameters.dynamic.comment_author;
+        icon = creatorIcon(channelUrl, notification_parameters?.dynamic?.comment_author_thumbnail);
+        break;
+      case RULE.COMMENT_REPLY:
+        channelUrl = notification_parameters.dynamic.reply_author;
+        icon = creatorIcon(channelUrl, notification_parameters?.dynamic?.comment_author_thumbnail);
+        break;
+      case RULE.NEW_CONTENT:
+        channelUrl = notification_parameters.dynamic.channel_url;
+        icon = creatorIcon(channelUrl, notification_parameters?.dynamic?.channel_thumbnail);
```

Comment:
DRY: the next person updating this might not know there is another duplicate to edit.

Usually, the refactoring goes into `util/notifications.js`, but this also involves jsx, so we might need to export from `component/notification/view.jsx`.